hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -195,13 +202,19 @@ public void setStateVersion(int stateVersion) {
 	@Override
 	public MerkleNetworkContext copy() {
 		setImmutable(true);
+
+		List<DeterministicThrottle.UsageSnapshot> copyUsageSnapshots = new ArrayList<>();
+		for(DeterministicThrottle.UsageSnapshot snapshot: usageSnapshots){
+			copyUsageSnapshots.add(new DeterministicThrottle.UsageSnapshot(snapshot.used(), snapshot.lastDecisionTime()));
+		}
+
 		return new MerkleNetworkContext(
 				consensusTimeOfLastHandledTxn,
 				seqNo.copy(),
 				lastScannedEntity,
 				midnightRates.copy(),
-				usageSnapshots,
-				congestionLevelStarts,
+				copyUsageSnapshots,
+				congestionLevelStarts.clone(),
```

Comment:
This is very close to the right idea! 👍 

However, we want the **immutable** `MerkleNetworkContext` to get the cloned array here (so instead of passing `congestionLevelStarts.clone()` to the copy, this instance should keep the clone).